### ISSUE TITLE
perldoc: S<...> does not work if =encoding is present

### ISSUE THREAD

**Issue by jimav** [Perl/perl5#19174]
In pod documentation, if any **=encoding** statement is present then the `S<nonbreakable text>` feature stops working, i.e. the enclosed text is not protected from line-wrapping, at least when rendered on a terminal with 'perldoc'.

STEPS TO REPRODUCE:
1. Put the following test data into a file, say, test.pod
2. Open a terminal window with 80 columns (either xterm or gnome-terminal)
3. perldoc test.pod

RESULTS:

The "use My::Wonderful::Package qw(:MYFEATURE)" appears broken across
two lines.   However if the =encoding statement is removed, then it
appears unbroken on a separate line, as it should.

```
=pod

=encoding ascii

=head1 THIS IS A HEADING

Please run perldoc on this file in an 80-column terminal.

This should be rendered on a single line:
S<use My::Wonderful::Package qw(:MYFEATURE)>

However the S < ... > does not work unless the "=encoding" line is removed.

=cut
```

**Perl configuration**
perldoc -V: Perldoc v3.2801, under perl v5.030000 for linux
Perl v5.30.0

perl -V: [perldashVoutput.txt](https://github.com/Perl/perl5/files/7305408/perldashV.txt)

**Comment by jkeenan** [comment]
> In pod documentation, if any **=encoding** statement is present then the `S<nonbreakable text>` feature stops working, i.e. the enclosed text is not protected from line-wrapping, at least when rendered on a terminal with 'perldoc'.
> 

I cannot reproduce your finding.
```
$ cat gh-19174-nonbreakable.pod 
=pod

=encoding ascii

=head1 THIS IS A HEADING

Please run perldoc on this file in an 80-column terminal.

This should be rendered on a single line:
S<use My::Wonderful::Package qw(:MYFEATURE)>

However the S < ... > does not work unless the "=encoding" line is removed.

=cut
```
On Ubuntu Linux, I have available Terminal and XTerm.  On each I issued the following command:
```
perldoc gh-19174-nonbreakable.pod
```
On Terminal I got:
![gh-19174-perldoc-in-terminal](https://user-images.githubusercontent.com/96524/136463343-bd7f721f-453e-4a8d-b074-0d4ea4b89587.png)

On XTerm I got:
![gh-19174-perldoc-in-xterm](https://user-images.githubusercontent.com/96524/136463403-654cb5c6-dc8b-4378-80ad-2ba947977bd2.png)
```
$ perl -v | head -2 | tail -1
This is perl 5, version 34, subversion 0 (v5.34.0) built for x86_64-linux
```

**Comment by jimav** [comment]
Very strange.  @jkeenan would you please try with PAGER=more  and no more options?

FWIW, I'm runnning Ubuntu 20.04

I tried eliminating unnecessary environment vars and still see the problem:
![jima-in-xterm](https://user-images.githubusercontent.com/1262272/136474239-b787943c-0627-4907-be6e-eb9868a104db.png)

**Comment by jkeenan** [comment]
I don't normally have any setting for C<PAGER>.  But when I called `PAGER=more perldoc gh-19174-nonbreakable.pod` I got the same results as before in both Terminal and XTerm, *i.e.,* the text within the `S<...>` tag was maintained on a single line, with the line breaking before that text.

**Comment by tonycoz** [comment]
This appears to have been fixed in perl 5.32.

The formatting is done by Pod::Text when I tested:

```
tony@venus:.../git/perl$ ~/perl/5.30.2-debug/bin/perldoc -D ../19174.pod 
Ending switch processing.  Args are [../19174.pod] with 0 errors.
Pagers: /usr/bin/less -R, more, less, pg, view, cat/home/tony/perl/5.30.2-debug/bin/perldoc => Pod::Perldoc v3.2801

Formatter class Pod::Perldoc::ToText version 3.28 successfully loaded!
Will format with the class Pod::Perldoc::ToText
Searching for ../19174.pod
Found as ../19174.pod
About to try calling /usr/bin/less -R /tmp/PNkNVQvXb8
```

Pod::Perldoc::ToText uses Pod::Text, looking at the version history of podlators:

```
podlators 4.13 (2019-12-25)
...
    [Pod::Text] Fix behavior of S<> with Unicode input to be consistent
    with behavior with a default encoding, namely treat all whitespace
    inside S<> as non-space characters and do not collapse it with
    adjacent whitespace.
```

Perl 5.30 includes 4.11, perl 5.32 includes 4.14:
```
$ ./perl -Ilib utils/corelist -a Pod::Text | grep 5.3[02]
  v5.30.0    4.11      
  v5.30.1    4.11      
  v5.30.2    4.11      
  v5.30.3    4.11      
  v5.32.0    4.14      
  v5.32.1    4.14      
```

which explains why it works for @jkeenan and doesn't for @jimav.

Try upgrading to the latest podlators, I expect that will fix it for you.
